README.md
# Reports

**INACTIVE NOTICE: This plugin is unsupported by WPMUDEV, we've published it here for those technical types who might want to fork and maintain it for their needs.**

## Translations

Translation files can be found at https://github.com/wpmudev/translations

## With Reports you can monitor post, page and comment activity for a user or a single site across an entire network.

Add a page that lets you generate a wide variety of reports. Easily monitor activity for the past 15, 30 or 45 days.

### Generate Reports with Ease

All you need to do is select the report you want to create, add the username or site ID, the time period and then click View. 

![image](http://premium.wpmudev.org/wp-content/uploads/2008/11/reports63.jpg)

 Quickly create reports of any user's activity

### Display Clean Graphs

This plugin displays data in easy-to-read graphs making it simple to process information quickly. 

![image](http://premium.wpmudev.org/wp-content/uploads/2008/11/reports62.jpg)

 View comment activity on all sites across an entire network

 Keep an eye on site and user activity across an entire network with Reports.

## Usage

For help with installing plugins please see our [Plugin installation guide](https://premium.wpmudev.org/wpmu-manual/installing-regular-plugins-on-wpmu/). Once installed log into to your admin panel, visit **Network Admin -> Plugins** and **Network Activate** the plugin.

### Please Note:

If you are using Multi-DB and the auto-install does not add the db table(s) for you then run the [sql code in "sql.txt" on your database](https://premium.wpmudev.org/wpmu-manual/running-a-sql-code-on-your-wpmu-database/).

### To use:

1.  Go to **Network Admin > Settings > Reports** 

2.  Click on **View Report** next to the report you want to run.

![image](https://premium.wpmudev.org/wp-content/uploads/2008/11/reports61.jpg)

3.  Add the Site ID or the username, select the date range and click **View.** 

![image](https://premium.wpmudev.org/wp-content/uploads/2008/11/reports62.jpg)


changelog.txt
Plugin Name: Reports
Author: Andrew Billits, Ulrich Sossou, Ignacio Cruz

Change Log:
----------------------------------------------------------------------
----------------------------------------------------------------------

1.0.8 - 02/02/2015
----------------------------------------------------------------------
- Fixed: Several Stric/Warnings Notifications.

1.0.7 - 05/01/2013
----------------------------------------------------------------------
- Fixed little XSS issue 

1.0.6 - 04/26/2013
----------------------------------------------------------------------
- Fixed SQL Injection bugs
- Disabled user activity recording
- Fixed XSS issue 

1.0.5 - 02/24/2011
----------------------------------------------------------------------
- Singlesite support

1.0.4 - 01/10/2011
----------------------------------------------------------------------
- 3.1+ compatibility update
- Internationalization

1.0.3 - 06/01/2010
----------------------------------------------------------------------
- 3.0+ compatibility update

1.0.2 - 03/09/2010
----------------------------------------------------------------------
- Fixed typo in sql.txt

1.0.1 - 03/02/2009
----------------------------------------------------------------------
- Added site admin check.

1.0.0 - 11/20/2008
----------------------------------------------------------------------
- Initial Release.


reports/user_comments.php
<?php
/*
Plugin Name: Report - User Comments
Plugin URI: 
Description:
Author: Andrew Billits
Version: 1.0.1
Author URI:
*/

/* 
Copyright 2007-2009 Incsub (http://incsub.com)

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (Version 2 - GPLv2) as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

//------------------------------------------------------------------------//
//---Hook-----------------------------------------------------------------//
//------------------------------------------------------------------------//

reports_add_report('User Comments','user-comments','Displays comment activity for a user');

if ( $_GET['report'] == 'user-comments' ) {
	add_action('view_report','report_user_comments_ouput');
}

//------------------------------------------------------------------------//
//---Outpur Functions-----------------------------------------------------//
//------------------------------------------------------------------------//

function report_user_comments_ouput(){
	global $wpdb, $current_site;
	switch( $_GET[ 'report-action' ] ) {
		//---------------------------------------------------//
		default:
			?>
			<form name="report" method="post" action="ms-admin.php?page=reports&action=view-report&report=user-comments&report-action=view">
				<table class="form-table">
				<tr valign="top">
				<th scope="row"><?php _e('Username') ?></th>
				<td><input type="text" name="user_login" id="user_login" style="width: 95%" tabindex='1' maxlength="200" value="" />
				<br />
				<?php _e('Case Sensative') ?></td> 
				</tr>
				<tr valign="top">
				<th scope="row"><?php _e('Period') ?></th>
				<td>
				<select name="period" id="period">
					<option value="15" ><?php _e('15 Days'); ?></option>
					<option value="30" ><?php _e('30 Days'); ?></option>
					<option value="45" ><?php _e('45 Days'); ?></option>
				</select>
				<br /><?php //_e('') ?></td>
				</tr>
				</table>
			<p class="submit">
			<input type="submit" name="Submit" value="<?php _e('View') ?>" />
			</p>
			</form>
			<?php
		break;
		//---------------------------------------------------//
		case "view":
			$user_count = $wpdb->get_var("SELECT COUNT(*) FROM " . $wpdb->base_prefix . "users WHERE user_login = '" . $_POST['user_login'] . "'");
			if ( $user_count == '0' ) {
				?>
                <p><?php _e('User not found.'); ?></p>
                <?php
			} else {
				$user_ID = $wpdb->get_var("SELECT ID FROM " . $wpdb->base_prefix . "users WHERE user_login = '" . $_POST['user_login'] . "'");
				?>
                <p>
                    <ul>
                        <li><strong><?php _e('Username'); ?></strong>: <?php echo $_POST['user_login']; ?></li>
                        <li><strong><?php _e('Period'); ?></strong>: <?php echo __($_POST['period'] . ' ' . 'Days'); ?></li>
                    </ul>
                </p>
                <?php
				//=======================================//
				$report_data = array();
				$days = 0;
				$total_days = $_POST['period'];
				$total_days_safe = $_POST['period'] + 3;
				$date_format = get_option('date_format');

				$query = "SELECT DATE_FORMAT( date_time, '%Y-%m-%d' ) as formatted_date FROM " . $wpdb->base_prefix . "reports_comment_activity WHERE user_ID = '" . $user_ID . "' AND date_time > '" . reports_days_ago($total_days_safe,'Y-m-d') . " 00:00:00'";
				$report_results = $wpdb->get_results( $query, ARRAY_A );
				while ( $days <= $total_days ) {
					$count = 0;
					$value = 0;
					$day = reports_days_ago($days,'Y-m-d');
					if ( count( $report_results ) > 0 ) {
						foreach ( $report_results as $report_result ) {
							if ($report_result['formatted_date'] == $day) {
								$count = $count + 1;
							}
						}
					}
					$label = reports_days_ago($days,$date_format);
					$value = $count;
					$report_data[] = array($label,$value);
					$days = $days + 1;
				}
				
				$report_data = array_reverse($report_data);
				//=======================================//
				include_once(ABSPATH . 'wp-content/report-graphs/open-flash-chart/open-flash-chart.php');
				$count = 0;
				$array_labels = array();
				$array_values = array();
				$piwik_api_response = array();
				$piwik_api_response[] = array('1','2');
				foreach ( $report_data as $array_item ) {
					$count = $count + 1;
					if ( $count != 1 ) {
						$array_labels[] = $array_item[0];
						$array_values[] = $array_item[1];
					}
				}
				$label_count = count( $array_labels );
				$highest_value = 0;
				foreach ( $array_values as $value ) {
					if ( $value > $highest_value) {
						$highest_value = $value;
					}
				}
				//=======================================//
				$g = new graph();
				//------------------------------//
				//---Data-----------------------//
				//------------------------------//
				$g->set_data( $array_values );
				
				//------------------------------//
				//---X--------------------------//
				//------------------------------//
				$g->set_x_labels( $array_labels );
				//------------------------------//
				//---Y--------------------------//
				//------------------------------//
				$g->set_y_min( 0 );
				$g->set_y_max( $highest_value );
				//------------------------------//
				$g->set_num_decimals ( 0 );
				$g->set_is_decimal_separator_comma( false );
				$g->set_is_thousand_separator_disabled( true );  
				$g->y_axis_colour = '#ffffff';
				$g->x_axis_colour = '#596171'; 
				$g->x_grid_colour = $g->y_grid_colour = '#E0E1E4';
				
				// approx 5 x labels on the graph
				$steps = ceil($label_count / 5);
				$steps = $steps + $steps % 2; // make sure modulo 2
				
				$g->set_x_label_style( 10, $g->x_axis_colour, 0, $steps, $g->x_grid_colour );
				$g->set_x_axis_steps( $steps / 2 );
				
				
				$stepsY = ceil($highest_value / 4);
				$g->y_label_steps( $stepsY / 3 );
				$g->y_label_steps( 4 );
				
				$g->bg_colour = '#ffffff';
				$g->set_inner_background('#ffffff');
				$g->area_hollow(1,3,4,'#3357A0', __('comment(s)'),10);
				
				$g->set_tool_tip( '#x_label# <br>#val# #key# ' );
				//------------------------------//
				$g->set_width( '100%' );
				$g->set_height( 250 );
				$g->set_js_path ( 'http://' . $current_site->domain . $current_site->path . 'wp-content/report-graphs/open-flash-chart/js/' );
				$g->set_swf_path ( 'http://' . $current_site->domain . $current_site->path . 'wp-content/report-graphs/open-flash-chart/' );
				$g->set_output_type('js');
				echo $g->render();
				//=======================================//
			}
		break;
		//---------------------------------------------------//
	}
}

?>
